lib/slack.ml

```diff
@@ -267,13 +256,43 @@ let generate_status_notification (cfg : Config_t.config) (notification : status_
       ]
   in
   let summary =
+    let state_info =
+      match state with
+      | Success -> "succeeded"
+      | Failure -> "failed"
+      | Error -> "error"
+      | _ ->
+        invalid_arg
+          (sprintf "Monorobot doesn't know how to generate notification for the unexpected event %s of %s"
+             (string_of_status_state state) sha
+          )
+    in
+    let commit_message = first_line message in
     match target_url with
     | None ->
-      sprintf "<%s|[%s]> CI Build Status notification: %s" repository.url repository.full_name state_info
+      sprintf "<%s|[%s]>: Build %s for \"%s\"" repository.url repository.full_name state_info commit_message
       (* in case the CI run is not using buildkite *)
     | Some target_url ->
-      sprintf "<%s|[%s]> CI Build Status notification for <%s|%s>: %s" repository.url repository.full_name target_url
-        context state_info
+      (* try to compute the pipeline_url based on the `context` (e.g. buildkite/pipeline) and the `target_url` (link to the build) *)
+      let pipeline_url =
+        let pipeline_name = context |> String.split ~on:'/' |> List.last in
```

Comment:
I think that it won't work when the context is for a single step, as it would be `orgname/pipelinename/jobname`

Reply:
Oh I see, `context` does not always have 2 "components". Looking at more data examples, I found other cases with 1 and 3 components: 

*  `"context": "buildkite/pipeline2/build"`
* `"context": "default",

It shouldn't be an issue anymore with [`5024756` (#142)](https://github.com/ahrefs/monorobot/pull/142/commits/502475650b620cab2d049a3b828e2f0a3b188135)